packages/react-app/src/views/QuadraticDiplomacyVote.jsx

```diff
@@ -1,176 +1,234 @@
-import { Button, Checkbox, Divider, Space } from "antd";
-import React, {useEffect, useState} from "react"
+import React, { useEffect, useState, useMemo, useCallback } from "react";
+import { Button, Checkbox, Divider, Space, List, Steps, Typography, Badge } from "antd";
+import { SmileTwoTone, LikeTwoTone, CheckCircleTwoTone, MinusOutlined, PlusOutlined } from "@ant-design/icons";
+import { Address } from "../components";
+const { Title } = Typography;
 
-export default function QuadraticDiplomacyVote({
-  voteCredits,
-  contributorEntries,
-  tx,
-  writeContracts,
-}) {
-  const [contributors, setContributors] = useState({});
+export default function QuadraticDiplomacyVote({ voteCredits, contributorEntries, tx, writeContracts }) {
+  // local state
   const [selectedContributors, setSelectedContributors] = useState({});
-  const [currentPage, setCurrentPage] = useState(1);
+  const [currentStep, setCurrentStep] = useState(0);
   const [availableVoteTokens, setAvailableVoteTokens] = useState(0);
 
-  useEffect(async () => {
-    const initialContributors = contributorEntries.reduce((entries, current) => {
-      entries[current.wallet] = {
-        name: current.name,
-        voteTokens: 0,
-      }
-
-      return entries;
-    }, {})
-
-    setContributors(initialContributors);
-  }, [contributorEntries]);
+  // memoized values
+  const contributors = useMemo(
+    () =>
+      contributorEntries.reduce((entries, current) => {
+        entries[current.wallet] = {
+          name: current.name,
+          voteTokens: 0,
+        };
+        return entries;
+      }, {}),
+    [contributorEntries],
+  );
+  const allcontributorsSelected = useMemo(
+    () => Object.keys(contributors).length === Object.keys(selectedContributors).length,
+    [contributors, selectedContributors],
+  );
```

Comment:
This is the only `useMemo` that I believe does not add value. Also we should use camelCase right? `allContributorsSelected `

Reply:
nice catch. I wonder why ESlint did not thrown me a warning for this.